internal/grpc/services/gateway/storageprovider.go

```diff
@@ -1268,28 +1273,22 @@ func (s *svc) ListContainer(ctx context.Context, req *provider.ListContainerRequ
 	panic("gateway: stating an unknown path:" + p)
 }
 
-func (s *svc) getPath(ctx context.Context, ref *provider.Reference, keys ...string) (string, error) {
+func (s *svc) getPath(ctx context.Context, ref *provider.Reference, keys ...string) (string, *rpc.Status) {
 	if ref.GetPath() != "" {
-		return ref.GetPath(), nil
+		return ref.GetPath(), &rpc.Status{Code: rpc.Code_CODE_OK}
 	}
 
 	if ref.GetId() != nil && ref.GetId().GetOpaqueId() != "" {
 		req := &provider.StatRequest{Ref: ref, ArbitraryMetadataKeys: keys}
 		res, err := s.stat(ctx, req)
-		if err != nil {
-			err = errors.Wrap(err, "gateway: error stating ref:"+ref.String())
-			return "", err
-		}
-
-		if res.Status.Code != rpc.Code_CODE_OK {
-			err := status.NewErrorFromCode(res.Status.Code, "gateway")
-			return "", err
+		if (res != nil && res.Status.Code != rpc.Code_CODE_OK) || err != nil {
```

Comment:
Do we need to check `res != nil`? There might arise cases where we might have initialized it and then run into some error but returned it anyway.